atom/browser/ui/win/thumbar_host.cc

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2015 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "atom/browser/ui/win/thumbar_host.h"
+
+#include <shobjidl.h>
+
+#include "base/win/scoped_comptr.h"
+#include "base/win/win_util.h"
+#include "base/win/wrapped_window_proc.h"
+#include "base/strings/utf_string_conversions.h"
+#include "third_party/skia/include/core/SkBitmap.h"
+#include "ui/gfx/icon_util.h"
+#include "ui/gfx/win/hwnd_util.h"
+
+namespace atom {
+
+namespace {
+
+// From MSDN: https://msdn.microsoft.com/en-us/library/windows/desktop/
+// dd378460(v=vs.85).aspx#thumbbars
+// The thumbnail toolbar has a maximum of seven buttons due to the limited room.
```

Comment:
We do need to do it here. If we pass > 7 buttons to Windows API directly, nothing will happen(The thumbnail bar will totally not show).